examples/service-catalog/service-catalog.yaml

```diff
@@ -0,0 +1,370 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: service-catalog
+objects:
+
+- kind: ClusterRoleBinding
```

Comment:
> clusterrole/binding allowing anyone to call the template broker (needed until we have auth support for SC->broker flows)

Doesn't the TSB trust callers to provide information about who is requesting a particular resource?  I remember having a special permission to allow psuedo-impersonation for that purpose, but if the SC lacks the ability to authenticate, how are you going to restrict it?



> clusterrole/binding allowing the SC SA to view all namespaces

Fine.  This should be stock eventually.



> a clusterrole/binding allowing the SC SA to update broker/instance/binding status

Why is necessary?  The SC should have a direct (never touching aggregator) loopback connection to itself which has full permissions.



> a role/binding allowing the SC SA to mange endpoints in its project (needed for etcd for now, should be removed eventually)

Are you exposing etcd as as service in your namespace?  I did not expect that.  Doesn't that open etcd to attack from anyone on the cluster?





> Assuming 5+6 should be namespaced for users, i'll need direction on how we make that happen (ie how to ensure users get the role bound when they create new projects).

It sounds like it should be namespace scoped.  It can be done in a project template.  Be sure that you don't forget an optional `oadm migrate` step.